packages/buidler-core/src/internal/buidler-evm/stack-traces/solidityTracer.ts

```diff
@@ -1099,9 +1124,14 @@ export class SolidityTracer {
       }
 
       const inst = trace.bytecode.getInstruction(step.pc);
+
       if (inst.location !== undefined) {
         return i;
       }
+
+      if (stopOnLastJump && inst.opcode === Opcode.JUMPI) {
```

Comment:
Why `JUMPI`?

Reply:
I think you want `inst.jumpType === JumpType.INTO_FUNCTION`

Reply:
I'm... not really sure, I'm pretty much playing it by ear here. But if I change it to `inst.jumpType === JumpType.INTO_FUNCTION` there are five tests that stop working:

- `0_6/call-message/external-calls/non-contract-account-called`
- `0_6/create-message/external-calls/non-contract-account-called`
- `0_6/create-message/no-calls/nonpayable-with-wrong-args`
- `0_6/create-message/no-calls/payable-with-wrong-args`
- `0_6/create-message/no-calls/payable-with-wrong-args-and-value`

To use the last one as example, the test expects a `INVALID_PARAMS_ERROR` but gets a `REVERT_ERROR` in the constructor instead. The last part of the trace looks like this:

```
    ...
    019   ADD                                      c.sol:C:   -  41:40
    020   PUSH1 0x40                               c.sol:C:   -  41:40
    022   MSTORE                                   c.sol:C:   -  41:40
    023   PUSH1 0x20                               
    025   DUP2                                     
    026   LT                                       
    027   ISZERO                                   
    028   PUSH1 0x23                               
    030   JUMPI (INTERNAL_JUMP)            
    031   PUSH1 0x00                               
    033   DUP1                                     
    034   REVERT                                   
```